lib/checker.js

```diff
@@ -60,15 +93,22 @@ Checker.prototype.checkDirectory = function(path) {
             }
 
             return vowFs.stat(fullname).then(function(stat) {
+                var promise;
+
                 if (stat.isDirectory()) {
                     return this.checkDirectory(fullname);
                 }
 
                 if (!this._hasCorrectExtension(fullname)) {
-                    return [];
+                    if (this._configuration.getExtract() === 'never') {
+                        return [];
+                    }
+                    promise = Vow.when(this.extractFile(fullname));
```

Comment:
How about extending defaults of `fileExtensions` option instead of introducing the `extract` and putting every possible html extension?

It has obvious downside, like if user redefine it, their would need to add those extension again,